app/components/layout.js

```diff
@@ -68,16 +66,22 @@ class Layout extends React.Component {
           </div>
         </div>
 
-        <Button onClick={this.connectToServer} value='Connect To Server' />
+        <Button onClick={this.connectToServer.bind(this)} value={this.props.connectionStatus} />
+
+        <Table />
       </div>
     )
   }
 }
 
-Layout.propTypes = {
-  // Validation logic
-};
+function mapDispatchToProps(dispatch) {
+  return {
+    actions: bindActionCreators(actions, dispatch)
+  }
+}
 
-window.onload = function(){
-  ReactDOM.render(<Layout/>, document.getElementById('layout-container'));
+function mapStateToProps(state) {
+  return state
 }
+
+export default connect(mapStateToProps, mapDispatchToProps)(Layout)
```

Comment:
@kgrz Yup `connect` is a curried function. And this exporting derived help while reusing the components. otherwise the mapToState need to be redefined where ever we use this component.